pkgs/development/python-modules/osmpythontools/default.nix

```
@@ -0,0 +1,48 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, beautifulsoup4
+, geojson
+, lxml
+, matplotlib
+, numpy
+, pandas
+, ujson
+, xarray
+}:
+
+buildPythonPackage rec {
+  pname = "osmpythontools";
+  version = "0.2.6";
+
+  src = fetchPypi {
+    pname = "OSMPythonTools";
+    inherit version;
+    sha256 = "efc72e3963971c6c7fd94bd374704a5b78eb6c07397a4ffb5f9176c1e4aee096";
+  };
+
+  propagatedBuildInputs = [
+    beautifulsoup4
+    geojson
+    lxml
+    matplotlib
+    numpy
+    pandas
+    ujson
+    xarray
+  ];
+
+  patches = [ ./remove-unused-dependency.patch ];
+
```

half measure since upstream doesn't have any unit tests :(
```suggestion
  pythonImportsCheck = [
    "OSMPythonTools"
    "OSMPythonTools.api"
    "OSMPythonTools.data"
    "OSMPythonTools.element"
    "OSMPythonTools.nominatim"
    "OSMPythonTools.overpass"
    ];
```

```
running build_ext

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
Finished executing setuptoolsCheckPhase
derivation '/nix/store/an9gln3zivcplvd2y3z74fm3l67rrhz7-python3.7-osmpythontools-0.2.6.drv
```

---

Hmm ... that causes an error on my machine. Anything I'm doing wrong?

### Steps to reproduce

1. Change into a new directory 
   ```bash
   cd $(mktemp -d)
   ```
2. create a `shell.nix` that pulls in this derivation
   ```bash
   cat << EOF > shell.nix
   { pkgs ? import <nixpkgs> {} }:
   pkgs.mkShell {
     buildInputs = with pkgs; [
       (python3.withPackages (ps: [
         ps.osmpythontools
       ]))
     ];
   }
   EOF
   ```
3. try to launch a shell based on that
   ```bash
   nix-shell -I nixpkgs=$PATH_TO_NIXPKGS_CHECKOUT
   ```
   where e93faf67a877f2baa56dc1dfc532393d1002ad54 is checked out at `$PATH_TO_NIXPKGS_CHECKOUT`

#### expected

environment in which `python3` is available and within Python 3, `OSMPythonTools` can be imported

#### observed

Build fails with `FileNotFoundError: [Errno 2] No such file or directory: 'setup.py'` and exit code `100`.

<details><summary>full terminal output (click to expand)</summary>

```
these derivations will be built:
  /nix/store/fsc0bi90z8vsfrjv72xawfhl1vb6y5zb-python3.7-osmpythontools-0.2.6.drv
  /nix/store/7k4akh88yjppgsjppig3papyy9sd88q3-python3-3.7.6-env.drv
building '/nix/store/fsc0bi90z8vsfrjv72xawfhl1vb6y5zb-python3.7-osmpythontools-0.2.6.drv'...
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing setuptools-check-hook
Using setuptoolsCheckPhase
unpacking sources
unpacking source archive /nix/store/jwl9s5qyg7rb30xfs3v1mhlkc2y0ixzf-OSMPythonTools-0.2.6.tar.gz
source root is OSMPythonTools-0.2.6
setting SOURCE_DATE_EPOCH to timestamp 1579135072 of file OSMPythonTools-0.2.6/setup.cfg
patching sources
applying patch /nix/store/651gwb9kd2bhzi0dh34h9x5v272kaz70-remove-unused-dependency.patch
patching file OSMPythonTools.egg-info/requires.txt
patching file setup.py
configuring
no configure script, doing nothing
building
Executing setuptoolsBuildPhase
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/OSMPythonTools
copying OSMPythonTools/__info__.py -> build/lib/OSMPythonTools
copying OSMPythonTools/overpass.py -> build/lib/OSMPythonTools
copying OSMPythonTools/__init__.py -> build/lib/OSMPythonTools
copying OSMPythonTools/element.py -> build/lib/OSMPythonTools
copying OSMPythonTools/api.py -> build/lib/OSMPythonTools
copying OSMPythonTools/nominatim.py -> build/lib/OSMPythonTools
copying OSMPythonTools/data.py -> build/lib/OSMPythonTools
creating build/lib/OSMPythonTools/internal
copying OSMPythonTools/internal/cacheObject.py -> build/lib/OSMPythonTools/internal
copying OSMPythonTools/internal/__init__.py -> build/lib/OSMPythonTools/internal
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/OSMPythonTools
copying build/lib/OSMPythonTools/__info__.py -> build/bdist.linux-x86_64/wheel/OSMPythonTools
copying build/lib/OSMPythonTools/overpass.py -> build/bdist.linux-x86_64/wheel/OSMPythonTools
copying build/lib/OSMPythonTools/__init__.py -> build/bdist.linux-x86_64/wheel/OSMPythonTools
copying build/lib/OSMPythonTools/element.py -> build/bdist.linux-x86_64/wheel/OSMPythonTools
copying build/lib/OSMPythonTools/api.py -> build/bdist.linux-x86_64/wheel/OSMPythonTools
copying build/lib/OSMPythonTools/nominatim.py -> build/bdist.linux-x86_64/wheel/OSMPythonTools
copying build/lib/OSMPythonTools/data.py -> build/bdist.linux-x86_64/wheel/OSMPythonTools
creating build/bdist.linux-x86_64/wheel/OSMPythonTools/internal
copying build/lib/OSMPythonTools/internal/cacheObject.py -> build/bdist.linux-x86_64/wheel/OSMPythonTools/internal
copying build/lib/OSMPythonTools/internal/__init__.py -> build/bdist.linux-x86_64/wheel/OSMPythonTools/internal
running install_egg_info
running egg_info
writing OSMPythonTools.egg-info/PKG-INFO
writing dependency_links to OSMPythonTools.egg-info/dependency_links.txt
writing requirements to OSMPythonTools.egg-info/requires.txt
writing top-level names to OSMPythonTools.egg-info/top_level.txt
reading manifest file 'OSMPythonTools.egg-info/SOURCES.txt'
writing manifest file 'OSMPythonTools.egg-info/SOURCES.txt'
Copying OSMPythonTools.egg-info to build/bdist.linux-x86_64/wheel/OSMPythonTools-0.2.6-py3.7.egg-info
running install_scripts
creating build/bdist.linux-x86_64/wheel/OSMPythonTools-0.2.6.dist-info/WHEEL
creating 'dist/OSMPythonTools-0.2.6-py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'OSMPythonTools/__info__.py'
adding 'OSMPythonTools/__init__.py'
adding 'OSMPythonTools/api.py'
adding 'OSMPythonTools/data.py'
adding 'OSMPythonTools/element.py'
adding 'OSMPythonTools/nominatim.py'
adding 'OSMPythonTools/overpass.py'
adding 'OSMPythonTools/internal/__init__.py'
adding 'OSMPythonTools/internal/cacheObject.py'
adding 'OSMPythonTools-0.2.6.dist-info/METADATA'
adding 'OSMPythonTools-0.2.6.dist-info/WHEEL'
adding 'OSMPythonTools-0.2.6.dist-info/top_level.txt'
adding 'OSMPythonTools-0.2.6.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Finished executing setuptoolsBuildPhase
installing
Executing pipInstallPhase
/build/OSMPythonTools-0.2.6/dist /build/OSMPythonTools-0.2.6
Processing ./OSMPythonTools-0.2.6-py3-none-any.whl
Requirement already satisfied: matplotlib in /nix/store/ja1qbgcqw9lfgi9cnjfx00qs2cldy5kj-python3.7-matplotlib-3.1.3/lib/python3.7/site-packages (from OSMPythonTools==0.2.6) (3.1.3)
Requirement already satisfied: beautifulsoup4 in /nix/store/n48h3i89a7sc9k1nq9fhfwpzil6k3jbx-python3.7-beautifulsoup4-4.8.2/lib/python3.7/site-packages (from OSMPythonTools==0.2.6) (4.8.2)
Requirement already satisfied: geojson in /nix/store/7b881yqpvv6j1w7rmfgx4rkli66h49l2-python3.7-geojson-2.5.0/lib/python3.7/site-packages (from OSMPythonTools==0.2.6) (2.5.0)
Requirement already satisfied: ujson in /nix/store/6s9zvjbx5nnfqz4j8i1n6fw1iwxx9sy5-python3.7-ujson-1.35/lib/python3.7/site-packages (from OSMPythonTools==0.2.6) (1.35)
Requirement already satisfied: xarray in /nix/store/6jpi9z1vp4ika5w6jp2xv8gs663mvqxz-python3.7-xarray-0.15.0/lib/python3.7/site-packages (from OSMPythonTools==0.2.6) (0.15.0)
Requirement already satisfied: pandas in /nix/store/qlc1nsmy69hssjh4kz7g160nbkajg0j9-python3.7-pandas-1.0.1/lib/python3.7/site-packages (from OSMPythonTools==0.2.6) (1.0.1)
Requirement already satisfied: numpy in /nix/store/67bz5c2pjyix6vvqkyksgn76kc43cal2-python3.7-numpy-1.18.1/lib/python3.7/site-packages (from OSMPythonTools==0.2.6) (1.18.1)
Requirement already satisfied: lxml in /nix/store/j3wxx0lwvz62rifirjd8fvr78cxv5sny-python3.7-lxml-4.5.0/lib/python3.7/site-packages (from OSMPythonTools==0.2.6) (4.5.0)
Requirement already satisfied: cycler>=0.10 in /nix/store/vrdhmmpmn123qqh2rc65y5r51120mq1m-python3.7-cycler-0.10.0/lib/python3.7/site-packages (from matplotlib->OSMPythonTools==0.2.6) (0.10.0)
Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in /nix/store/5gxcd90ag0psyj9xfw4pkgym5jw1r2h2-python3.7-pyparsing-2.4.6/lib/python3.7/site-packages (from matplotlib->OSMPythonTools==0.2.6) (2.4.6)
Requirement already satisfied: python-dateutil>=2.1 in /nix/store/j2kwbrb766fjkj4z40z0pccm9qbf9z8q-python3.7-python-dateutil-2.8.1/lib/python3.7/site-packages (from matplotlib->OSMPythonTools==0.2.6) (2.8.1)
Requirement already satisfied: kiwisolver>=1.0.1 in /nix/store/iy7kivzrqz6mvz09ipx5ag55dikm6bdf-python3.7-kiwisolver-1.1.0/lib/python3.7/site-packages (from matplotlib->OSMPythonTools==0.2.6) (1.1.0)
Requirement already satisfied: soupsieve>=1.2 in /nix/store/9lvgw21ggdyn6x29qpgbkf1kc114wq1k-python3.7-soupsieve-1.9.5/lib/python3.7/site-packages (from beautifulsoup4->OSMPythonTools==0.2.6) (1.9.5)
Requirement already satisfied: pytz>=2017.2 in /nix/store/6i5ds99fcily05i96l23km6in5mavi3r-python3.7-pytz-2019.3/lib/python3.7/site-packages (from pandas->OSMPythonTools==0.2.6) (2019.3)
Requirement already satisfied: six in /nix/store/p1ldp0xrkv9vywgray8r77jnavb12m2b-python3.7-six-1.14.0/lib/python3.7/site-packages (from cycler>=0.10->matplotlib->OSMPythonTools==0.2.6) (1.14.0)
Requirement already satisfied: setuptools in /nix/store/7zp4i556dz0m032mdwhqnkajm7mv6mrf-python3.7-setuptools-44.0.0/lib/python3.7/site-packages (from kiwisolver>=1.0.1->matplotlib->OSMPythonTools==0.2.6) (44.0.0.post20200211)
Installing collected packages: OSMPythonTools
Successfully installed OSMPythonTools-0.2.6
/build/OSMPythonTools-0.2.6
Finished executing pipInstallPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/rr1jp02bsz1yjhr9fms2crzwn9c82dc7-python3.7-osmpythontools-0.2.6
strip is /nix/store/xpzdkp9y8fnliwrs2g5pygx2qmmbrv8w-binutils-2.31.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/rr1jp02bsz1yjhr9fms2crzwn9c82dc7-python3.7-osmpythontools-0.2.6/lib
patching script interpreter paths in /nix/store/rr1jp02bsz1yjhr9fms2crzwn9c82dc7-python3.7-osmpythontools-0.2.6
checking for references to /build/ in /nix/store/rr1jp02bsz1yjhr9fms2crzwn9c82dc7-python3.7-osmpythontools-0.2.6...
running install tests
no Makefile or custom buildPhase, doing nothing
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: OSMPythonTools OSMPythonTools.api OSMPythonTools.data OSMPythonTools.element OSMPythonTools.nominatim OSMPythonTools.overpass
setuptoolsCheckPhase
Executing setuptoolsCheckPhase
Traceback (most recent call last):
  File "nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "/nix/store/g20g00saxyssz0ka1xcmcm31bfq6iq31-python3-3.7.6/lib/python3.7/tokenize.py", line 447, in open
    buffer = _builtin_open(filename, 'rb')
FileNotFoundError: [Errno 2] No such file or directory: 'setup.py'
builder for '/nix/store/fsc0bi90z8vsfrjv72xawfhl1vb6y5zb-python3.7-osmpythontools-0.2.6.drv' failed with exit code 1
cannot build derivation '/nix/store/7k4akh88yjppgsjppig3papyy9sd88q3-python3-3.7.6-env.drv': 1 dependencies couldn't be built
error: build of '/nix/store/7k4akh88yjppgsjppig3papyy9sd88q3-python3-3.7.6-env.drv' failed
```

</details>

### system information

(`nix-shell -p nix-info --run "nix-info -m"`)

 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.103, NixOS, 19.09.2079.8731aaaf8b3 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.2`
 - channels(root): `"nixos-19.09.2079.8731aaaf8b3"`
 - channels(das-g): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

---

that is expected behavior. The `setuptoolsBuildPhase` expects the setup.py to be in the PWD.

---

Also, the default checkPhase (setuptoolsCheckPhase). Is essentially `python setup.py test`, so setup.py needs to be IN PWD as well during that phase (unless you want to override)

---

actually, this is an oversight, I sent a PR https://github.com/NixOS/nixpkgs/pull/80595 to fix this.

---

included `doCheck = false;` in [`5ccb2e15`](https://github.com/NixOS/nixpkgs/compare/e93faf67a877f2baa56dc1dfc532393d1002ad54..5ccb2e150d854791167e5a6e560017327821c1dc)